src/shared/errors.ts

```diff
@@ -452,14 +466,16 @@ export class PermissionsError extends ToolkitError {
         public readonly userInfo: os.UserInfo<string>,
         public readonly expected: PermissionsTriplet
     ) {
-        const mode = modeToString(stats.mode)
+        const mode = `${stats.isDirectory() ? 'd' : '-'}${modeToString(stats.mode)}`
         const owner = stats.uid === userInfo.uid ? userInfo.username : stats.uid
-        const actual = modeToString(getEffectivePerms(userInfo.uid, userInfo.gid, stats)).slice(-3)
+        const { effective, isAmbiguous } = getEffectivePerms(userInfo.uid, userInfo.gid, stats)
+        const actual = modeToString(effective).slice(-3)
         const resolvedExpected = Array.from(expected)
             .map((c, i) => (c === '*' ? actual[i] : c))
             .join('')
+        const actualText = !isAmbiguous ? actual : `${mode.slice(-6, -3)} & ${mode.slice(-3)} (ambiguous)`
```

Comment:
If we actually try to read/write to the file/directory, does that resolve the ambiguity?